### ISSUE TITLE
Naming is hard, what about ArgChiseler ?

### ISSUE THREAD

**Issue by forax** [sormuras/command-line-interface#4]
Argument + Chisel => ArgChiseler

I like that the name refer to the cabinet wood working which is an art form more than something industrial.

**Comment by jbee** [comment]
Indeed. 
Too niche for my taste - given this should be something that everyone almost knows what it does on first contact the word chisel is too narrow and rarely used in programming or in general. 
For this we want something catchy but not fancy, I think.

**Comment by forax** [comment]
<bad faith>
You do not know the expression, fo chisel my nizzle :)
</bad faith>

**Comment by sormuras** [comment]
Naming is the hardest part.

That's why I started this repository with the long-ish but neutral `CommandLineInterface` name. Perhaps this was already the best idea for a utility that gets a `String[]` as an input and constructs an instance of user-defined `record` type - for the time being.

I guess, we need something that describes the "transformation of a structured string sequence into components of custom record declarations": `StructuredStringSequenceToRecordComponentsTransformer`

Suggestions for the first part of the name:

- Args
- Arguments
- Commands
- CommandLine
- Strings
- StringArray
- StringSequence
- StructuredSequence

Suggestions for the section and "active" part of the name:

- Evaluator
- Parser
- Processor
- Reader
- Splitter

**Comment by sormuras** [comment]
Another approach would to focus on the ouput part, the user-defined `record` type.

Suggestions for the first part of the name:

- Options
- Settings

Suggestions for the section and "active" part of the name:

- Builder
- Creator
- Constructor
- Producer
- Supplier

**Comment by forax** [comment]
`OptionParser` or `ArgumentParser` are ok for me.

Usually in English, you do not use a plural for the first part of the name.

**Comment by jbee** [comment]
Maybe not totally on point for this topic but when I started extracting the help part I had the thought  that the common part is that these both are utilities for handling arguments. 

One to map their string form to a record form (which includes deciding what they mean). The other one would be about printing a "man page" for them. Third one we talked about a bit would be validating arguments.  Following that thought it would be logical to use a naming pattern like _Argument(s){Function}_. 

For example:
* ArgumentDecoder / ArgumentAnalyser / ArgumentReader
* ArgumentDescriber / ArgumentPresenter / ArgumentWriter
* ArgumentValidator / ArgumentChecker / ArgumentVerifier

What these would share is the semantic model of what the different java types for the record component mean in terms of an argument. 

> You do not know the expression, fo chisel my nizzle :)

Clearly not :D And I start to wonder... is this safe to google? :grin: 

Usually I like a good metaphor but in this case I would go for boring and clear.

This is also why I dislike most of the suggestions made precisely because they are terms commonly used in programming to mean certain pattern or semantics and this is not any of them. And there is nothing worse than a name that creates a clear idea but which is quite wrong. Then it is better to chose a name which is unclear at first but gets clear with some javadoc reading.
